pkg/components/defaults.go

```diff
@@ -21,5 +21,6 @@ var (
 		Strip,
 		Header,
 		BasicAuth,
+		Auth,
```

Comment:
suggestion: I think calling this plugin auth is not specific enough. The intended usage is supporting authorization using _pre-authenticated and trusted headers_. It ultimately has nothing to do with authentication and is more similar to validation.

Reply:
I think I'll call it ValidateAuthHeaders to be more explicit